lambda/update/parse/changes.go

```diff
@@ -266,56 +332,98 @@ func oldEqualsExisting(old any, existing any) bool {
 //		s = append(s, v)
 //		return p.Consumed()
 //	})
-func (p *Parser) Each(fn func(int, *Parser) []shared.FieldError, required ...int) *Parser {
-	indexedChanges := map[int][]changeWithPosition{}
+func (p *Parser) Each(fn func(string, *Parser) []shared.FieldError, required ...int) *Parser {
```

Comment:
With the code as it is, `required` only works for `positionChanges` but I wanted some feedback before I went ahead with trying to get this working for both.

Reply:
If you put the new behaviour in an `EachKey(fn func(string, *Parser) []shared.FieldError) *Parser` method, you wouldn't need to support it. We only use `required` in one place, and that could continue to use the old `Each`.